### ISSUE TITLE
Clicking on a label does not trigger the change event on its control element.

### ISSUE THREAD

**Issue by mattiaskarlsson** [ariya/phantomjs#12032]
Clicking on a label does not trigger the change event on its control element. For example clicking on a label connected to an radio button input should check the radio button and trigger the change event but it doesn't.

phantomjs version 1.9.7

I've created a test page that should be able to reproduce the problem. It should print a "Changed" message in the console and create a sceenshot with the radio button checked.

**Test page:**

``` html
<!DOCTYPE html>
<html>
<head>
    <title>Label Bug Test</title>
    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
</head>
<body>
<form>
    <label id="label" for="input">label</label>
    <input type="radio" value="1" name="test" id="input"/>
</form>

<script>
    $(document).ready(function () {
        $('#input').on('change', function () {
            console.log("Changed");
        });
        $('#label').click();
    });
</script>
</body>
</html>
```

**Script to run the test:**

``` javascript
var page = require('webpage').create();
page.onConsoleMessage = function (msg) {
    console.log(msg);
};
page.open('bugtest.html', function () {
    page.render('result.png');
    phantom.exit();
});
```

**Comment by ef4** [comment]
It looks like this is fixed in 2.0.0.

**Comment by blimmer** [comment]
Can confirm. This is fixed for us, too.

**Comment by mattiaskarlsson** [comment]
Yes, this seems to work as expected in phantomjs 2.0.0 so we can probably close this bug.

**Comment by benkiefer** [comment]
Is there any possibility of a fix in this in the 1.9.x version. It's been over a year since this ticket was opened.

**Comment by 4096void** [comment]
sucks

**Comment by chouchou900822** [comment]
It's still have the problem in 2.0.0. To fix it, I have to create the click event by myself.
```js
page.evaluate(function () {
  var label = document.getElementById('label');
  var event = document.createEvent('MouseEvents');
  event.initMouseEvent('click', true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
  label.dispatchEvent(event);
});
```

**Comment by sascha-egerer** [comment]
Looks like the `focus` event of the field, that is related to the label, is not triggerd when the label is clicked. Does also not work on 2.1.1

**Comment by khebbie** [comment]
Hello!

 

You have a new message, please read <http://mega-data.biz/track.php?7nh>

 

klaus@hebsgaard.dk
